finagle-http/src/main/scala/com/twitter/finagle/http/DelayedReleaseService.scala

```diff
@@ -13,13 +27,38 @@ private[finagle] class DelayedReleaseService[-Req <: Request](
   service: Service[Req, Response]
 ) extends ServiceProxy[Req, Response](service) {
 
-  protected[this] val counter = new AsyncLatch
+  protected[this] val latch = new AsyncLatch
 
-  private[this] def proxy(in: Response) = {
+  override def apply(request: Req): Future[Response] = {
+    latch.incr()
+    service(request) transform {
+      case Return(r) if r.isChunked =>
+        Future.value(DelayedReleaseResponse(r, latch.decr))
```

Comment:
What do you think about moving `DelayedReleaseResponse.apply` here as `def private[this]`? This would not only reduce the diff, but would allow as to access `latch` directly (w/o passing a closure) so we can save those 8 bytes per closure allocation on each chunked request.